### PR TITLE
feat: Add Postgres support

### PR DESCRIPTION
**Pull request by Ant59**
This PR adds comprehensive Postgres support with the target of allowing easy import of MTGJSON data to a Hasura instance. This adds an `engine` flag to switch between Postgres and MySQL for `.sql` output files. I have tested with both clean Postgres and MySQL instances successfully.

### CODE REVIEW

File: mtgsqlive/__main__.py [hunk 1]
```diff
@@ -26,14 +26,27 @@
         metavar="fileOut",
     )
     parser.add_argument(
-        "--all",
-        help="Build all types (SQLite, SQL, CSV)",
+        "-a",
```
Comment (ZeldaZach) [inline mtgsqlive/__main__.py:31]: This is no longer backwards compatible. You can do "--all", "-a" to allow for both

Comment (Ant59) [reply mtgsqlive/__main__.py:31]: Should be all good now. I've made sure MySQL stays default and added long-hand and short-hand entries for all the flags so either will work, for consistency.

File: mtgsqlive/json2sql.py [hunk 2]
```diff
@@ -393,72 +535,87 @@ def get_sql_type(mixed, engine: str) -> str:
...
     for table_name, table_data in schema.items():
+        if engine == "postgres":
+            for attribute in sorted(table_data.keys()):
+                if "options" in table_data[attribute]:
+                    q += (
+                        f"CREATE TYPE {table_data[attribute]['type']} AS ENUM ('"
+                        + "', '".join(table_data[attribute]["options"])
+                        + "');\n"
+                    )
         q += f"CREATE TABLE `{table_name}` (\n"
```
Comment (Jaecen) [inline mtgsqlive/json2sql.py:549]: Postgres doesn't support backticks for identifiers. These need to be changed to quotes for Postgres. As-is, it generates SQL that `psql` fails to parse.

Comment (zebapy) [reply mtgsqlive/json2sql.py:549]: What's the parse error? I'm getting issue `ERROR:  null value in column "id" violates not-null constraint` using generated sql

**Comment by ZeldaZach** [comment]
Superseded by #84 :)

**Pull request closed without merging**